### ISSUE TITLE
text after auto page break doesn't keep margins

### ISSUE THREAD

**Issue by msakauye** [bpampuch/pdfmake#486]
Text values that get broken up by auto page break do not respect the defined margins on the new page(s).

![2016-01-14_17-15-47](https://cloud.githubusercontent.com/assets/6380278/12339332/872bd46c-bae2-11e5-89b0-c086620df215.png)

```
t = '';
for (var i = 0; i < 20; i++) {
    t += '('+i+') Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. ';
}

var dd = {
    content: [
        { text: t, style: 'main' },
    ],
    styles: {
        main: {
            margin: [40,0, 40, 0]
        },
    }
}
```

**Comment by oriolikart** [comment]
Hi @msakauye I have the same issue. Did you find any workarround/solution?

Thnx

**Comment by oriolikart** [comment]
The problem is when the page starts with 'style' that has margins defiend.

Try in playground : 

```
var dd = {
	content: [

		{ 
			text: [
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
								'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'This page starts WITH MARGIN\. This page starts WITH MARGIN. This page starts WITH MARGIN.'
				],
			style: 'headerWithMargin', 
			bold: false,
			pageBreak : "after"
		},
		{ 
			text: [
				'This page starts WITHOUT MARGIN.\n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Header style in this example sets alignment to justify, so this paragraph should be rendered \n',
				'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Malit profecta versatur nomine ocurreret multavit, officiis viveremus aeternum superstitio suspicor alia nostram, quando nostros congressus susceperant concederetur leguntur iam, vigiliae democritea tantopere causae, atilii plerumque ipsas potitur pertineant multis rem quaeri pro, legendum didicisse credere ex maluisset per videtis. Cur discordans praetereat aliae ruinae dirigentur orestem eodem, praetermittenda divinum. Collegisti, deteriora malint loquuntur officii cotidie finitas referri.'
				],
			style: 'headerWithOutMargin', 
			bold: false 
		}
	],
	styles: {
		headerWithMargin: {
			fontSize: 18,
			bold: true,
			alignment: 'justify',
			margin :  [40,0,0,0]
		},
		headerWithOutMargin: {
			fontSize: 18,
			bold: true,
			alignment: 'justify',
		}
	}
}
```

**Comment by massltov** [comment]
Temporary workaround in function DocumentContext.prototype.moveToNextPage change to:
`if (createNewPage) {
				var currentPageSize = getPageSize(this.getCurrentPage(), pageOrientation);
				var currentAvailableWidth = this.availableWidth;
				this.addPage(currentPageSize);
				this.availableWidth = currentAvailableWidth;
			} else {
				this.page = nextPageIndex;
				this.initializePage();
			}`

**Comment by massltov** [comment]
#353

**Comment by SIlver--** [comment]
+1 problem still exists, workaround works though

**Comment by liborm85** [comment]
Fixed by https://github.com/bpampuch/pdfmake/commit/287362a1b4763ee99fc0bb04061e12af9ee9e6d9

**Comment by RMstartup** [comment]
I am encountering issues with text not respecting margins on subsequent pages (other than the first).

I've confirmed that margins are 40 for all pages by debugging them in `initializePage` but as the screenshot shows, dates on the second page are farther right than on the first.

Left justification of the name and right justification of the date on the same line is achieved by setting the widths to '*' and 'auto' respectively in my js file.

I have confirmed I am using a pdfmake.js file that reflects fix that @liborm85 noted. Does anyone have suggestions for how I can resolve this?

![screen shot 2017-01-25 at 10 12 56 pm](https://cloud.githubusercontent.com/assets/7710026/22319425/9460f238-e350-11e6-8c0b-18dcb21a1739.png)
